crypto/types/multisig/threshold_pubkey.go

```diff
@@ -67,12 +74,65 @@ func (pk PubKeyMultisigThreshold) VerifyBytes(msg []byte, marshalledSig []byte)
 	return true
 }
 
+// VerifyMultisignature implements the PubKey.VerifyMultisignature method
+func (pk PubKeyMultisigThreshold) VerifyMultisignature(getSignBytes GetSignBytesFunc, sig *signing.MultiSignatureData) error {
+	bitarray := sig.BitArray
+	sigs := sig.Signatures
+	size := bitarray.Size()
+	// ensure bit array is the correct size
+	if len(pk.PubKeys) != size {
+		return fmt.Errorf("bit array size is incorrect %d", len(pk.PubKeys))
+	}
+	// ensure size of signature list
+	if len(sigs) < int(pk.K) || len(sigs) > size {
+		return fmt.Errorf("signature size is incorrect %d", len(sigs))
+	}
+	// ensure at least k signatures are set
+	if bitarray.NumTrueBitsBefore(size) < int(pk.K) {
+		return fmt.Errorf("")
+	}
+	// index in the list of signatures which we are concerned with.
+	sigIndex := 0
+	for i := 0; i < size; i++ {
+		if bitarray.GetIndex(i) {
+			si := sig.Signatures[sigIndex]
+			switch si := si.(type) {
+			case *signing.SingleSignatureData:
+				msg, err := getSignBytes(si.SignMode)
+				if err != nil {
+					return err
+				}
+				if !pk.PubKeys[i].VerifyBytes(msg, si.Signature) {
+					return err
+				}
+			case *signing.MultiSignatureData:
```

Comment:
according to [OctoLinker](https://octolinker.now.sh/), this case is not tested

Reply:
yeah, this is missing in coverage. Will add a test to cover